File: Objc in Swift app/DelegateInObjC/TaskManager.m
//
//  TaskManager.m
//  Objc in Swift app
//
//  Created by Cong Le on 2/1/24.
//


#import "TaskManager.h"

@implementation TaskManager
- (void)performTask {
    // Task is performed here
    NSString *result = @"Task Completed";
    
    // Notify the delegate that the task is completed
    if ([self.delegate respondsToSelector:@selector(taskManager:didCompleteTaskWithResult:)]) {
        [self.delegate taskManager:self didCompleteTaskWithResult:result];
    }
}
@end

File: Objc in Swift app/DelegationInSwift/TaskManager.swift
//
//  TaskManager.swift
//  Objc in Swift app
//
//  Created by Cong Le on 2/1/24.
//

import Foundation

protocol TaskDelegate: AnyObject {
    func didCompleteTask(_ sender: TaskManager, result: String)
}

class TaskManager {
    weak var delegate: TaskDelegate?     // Delegate property
    
    func performTask() {
        // Task is performed here
        let result = "Task Completed"
        
        // Notify the delegate that the task is completed
        delegate?.didCompleteTask(self, result: result)
    }
}
